envconfig.go

```diff
@@ -264,6 +264,12 @@ func processField(value string, field reflect.Value) error {
 		)
 		if field.Kind() == reflect.Int64 && typ.PkgPath() == "time" && typ.Name() == "Duration" {
 			var d time.Duration
+
+			// time.Parse does not support day durations, so we convert any day durations to an hours string
+			if newValue, changed := daysToHoursDurationStr(value); changed {
```

Comment:
Do we want to check this everytime? Could we do `strings.HasSuffix(value, "d")` to only run this when needed?